### PR TITLE
Add Cloud Spanner Batch DML sample

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: spanner/spanner_samples.rb [hunk 1]
```diff
@@ -968,6 +968,38 @@ def delete_using_partitioned_dml project_id:, instance_id:, database_id:
...
+def update_using_batch_dml project_id:, instance_id:, database_id:
+  # [START spanner_dml_batch_update]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: Add a new line here.

File: spanner/spanner_samples.rb [hunk 2]
```diff
@@ -968,6 +968,38 @@ def delete_using_partitioned_dml project_id:, instance_id:, database_id:
...
+  insert_statement = "INSERT INTO Albums "\
+    "(SingerId, AlbumId, AlbumTitle, MarketingBudget) "\
+    "VALUES (1, 3, 'Test Album Title', 10000)"
+  update_statement = "UPDATE Albums "\
+    "SET MarketingBudget = MarketingBudget * 2 "\
+    "WHERE SingerId = 1 and AlbumId = 3"
+  row_counts = Array.new
+  row_count = 0
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: Not needed as the value is defined below.

File: spanner/spanner_samples.rb [hunk 3]
```diff
@@ -968,6 +968,38 @@ def delete_using_partitioned_dml project_id:, instance_id:, database_id:
...
+  client.transaction do |transaction|
+    row_counts = transaction.batch_update do |b|
+      b.batch_update insert_statement
+      b.batch_update update_statement
+    end
+  end
+
+  row_count = row_counts.count
+
+  puts "Executed #{row_count} SQL statements using Batch DML."
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: If `rows` fits better use `rows.count` instead.

File: spanner/spanner_samples.rb [hunk 4]
```diff
@@ -968,6 +968,38 @@ def delete_using_partitioned_dml project_id:, instance_id:, database_id:
...
+    "(SingerId, AlbumId, AlbumTitle, MarketingBudget) "\
+    "VALUES (1, 3, 'Test Album Title', 10000)"
+  update_statement = "UPDATE Albums "\
+    "SET MarketingBudget = MarketingBudget * 2 "\
+    "WHERE SingerId = 1 and AlbumId = 3"
+  row_counts = Array.new
+  row_count = 0
+
+  client.transaction do |transaction|
+    row_counts = transaction.batch_update do |b|
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: Does `rows` fit better here as a name?

File: spanner/spanner_samples.rb [hunk 5]
```diff
@@ -968,6 +968,38 @@ def delete_using_partitioned_dml project_id:, instance_id:, database_id:
...
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+  insert_statement = "INSERT INTO Albums "\
+    "(SingerId, AlbumId, AlbumTitle, MarketingBudget) "\
+    "VALUES (1, 3, 'Test Album Title', 10000)"
+  update_statement = "UPDATE Albums "\
+    "SET MarketingBudget = MarketingBudget * 2 "\
+    "WHERE SingerId = 1 and AlbumId = 3"
+  row_counts = Array.new
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: Change to `row_counts = nil`
Keep it grouped together with `client.transaction do |transaction|`.

File: spanner/spanner_samples.rb [hunk 6]
```diff
@@ -968,6 +968,38 @@ def delete_using_partitioned_dml project_id:, instance_id:, database_id:
...
+  # [START spanner_dml_batch_update]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+  insert_statement = "INSERT INTO Albums "\
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: What about moving insert and update statements inside the transaction not using variables?

File: spanner/spanner_samples.rb [hunk 7]
```diff
@@ -1009,6 +1041,7 @@ def usage
   write_with_transaction_using_dml   <instance_id> <database_id> Update data using a DML statement within a read-write transaction.
   update_using_partitioned_dml       <instance_id> <database_id> Update multiple records using a partitioned DML statement.
   delete_using_partitioned_dml       <instance_id> <database_id> Delete multiple records using a partitioned DML statement.
+  update_using_batch_dml             <instance_id> <database_id> Updates sample data in the database using Batch DML.
```
Comment (frankyn) [inline spanner/spanner_samples.rb]: Please update the README.md with the new CLI command.

File: spanner/spanner_samples.rb (hunk 1 above)
Comment (jsimonweb) [reply spanner/spanner_samples.rb]: Done.

File: spanner/spanner_samples.rb (hunk 4 above)
Comment (jsimonweb) [reply spanner/spanner_samples.rb]: This variable contains an array of row counts updated or inserted by each DML transaction that is executed so I'd prefer to leave it as row_counts if that's acceptable.

File: spanner/spanner_samples.rb (hunk 7 above)
Comment (jsimonweb) [reply spanner/spanner_samples.rb]: Done.

File: spanner/spanner_samples.rb (hunk 4 above)
Comment (jsimonweb) [reply spanner/spanner_samples.rb]: I've renamed the variable previously name row_count to statement_count to make it more accurately named.

Comment (jsimonweb) [reply spanner/spanner_samples.rb]: * previously named row_count

File: spanner/spanner_samples.rb (hunk 6 above)
Comment (jsimonweb) [reply spanner/spanner_samples.rb]: Done.

File: spanner/spanner_samples.rb (hunk 5 above)
Comment (jsimonweb) [reply spanner/spanner_samples.rb]: Done.

**Comment by frankyn** [comment]
Thanks @jsimonweb, you can merge when you're ready.

**Pull request merged**